snapshot-labs/snapshot
# feat: add support for hibernated spaces

### Summary

Pitch https://github.com/orgs/snapshot-labs/projects/12/views/7?pane=issue&itemId=43161232

This PR add support for hibernated spaces:

- disable proposals creation on hibernated spaces
- show a hibernated warning for hibernated spaces on their main page + proposal creation page
- add option for space admins/controller to reactivate the space

### How to test

Install and use the following packages

- https://github.com/snapshot-labs/snapshot-sequencer/pull/197

A hibernation warning message should appear on the space all proposals page, as well as on the proposal creation page (same message for both)

The same warning message should be displayed for everyone, but the space admin/controller, who will see an additional button to reactivate the space (button link to space settings page)

1. Go to an hibernated space
2. A warning message should appear on the space's proposals page
3. Go to the create proposal page
4. The same message should appear, and the proposal creation button disabled
5. If you're admin of the hibernated spaces, you should see an additional button to reactivate it
6. Clicking on the "reactivate space" button should redirect to space settings page
7. A global warning message should always appear on all settings page
8. If the space does not have invalid settings, it should show a button to reactivate the space
9. If the space have invalid settings, it should show a message asking to fix the settings (a list of invalid settings also listed)
10. If all settings are valid, the reactivate space button should be enabled
11. Saving the settings should reactivate the space, and all warning message should disappear

### To-Do

- [x] Disable proposals creation for hibernated spaces
- [x] Show space warnings about hibernated space
- [x] Move text into translation files
- [x] Add link to hibernation doc
- [x] Add space reactivation logic

### Depends on

- [ ] https://github.com/snapshot-labs/snapshot-sequencer/pull/197 (to support receiving the `ReactivateSpace` action)
- [ ] OPTIONAL - https://github.com/snapshot-labs/snapshot-sequencer/pull/240 (to support reactivating spaces only when settings are valid)
